### ISSUE TITLE
Null Pointer Exception on repository.save

### ISSUE THREAD

**Issue by Kadavar32** [aerospike-community/spring-data-aerospike#385]
Hello, I am using spring-data-aerospike and found a problem with a specific use-case  when Aerospike Document is used to implement a lock, so lock document doesn't have any fields except `@Id` and `@Version`.

Since version 3.X.X key bin is not sent by default, and there are some conditions that lead to NullPointerException on `repository.save`

Here is a link of the code example with reproduced issue:
https://github.com/aerospike-community/spring-data-aerospike/compare/main...Kadavar32:feature/reproduce-npe-on-save

Conditions to be able to reproduce the issue:

1. Document with non writable fields only (id, expiration, version). 

```java
@Document
@RequiredArgsConstructor(onConstructor = @__({@PersistenceConstructor}))
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Data
@Builder
public class NoWritableFieldsDocument {

    @Id
    public final String id;

    @Version
    public long version;
}
```

2. Sending @class bin to the document is disabled. 

```java

@Bean
@Override
public AerospikeTypeAliasAccessor aerospikeTypeAliasAccessor() {
    return new AerospikeTypeAliasAccessor(null);
}
```

3. It doesn't matter if sending key bin is enabled or not (property `spring.aerospike.write.send-key=true|false`).

Repository example: 

```java
public interface NoWritableFieldsDocumentRepository extends AerospikeRepository<NoWritableFieldsDocument, String> {}
```

Test example: 

```java
@Test
public void shouldSaveDocumentWithNoWritableFields() {
    NoWritableFieldsDocument expectedDocument = NoWritableFieldsDocument.builder()
            .id("document-key-1")
            .build();

    noWritableFieldsDocumentRepository.save(expectedDocument);

    Optional<NoWritableFieldsDocument> actual = noWritableFieldsDocumentRepository.findById(expectedDocument.getId());

    assertThat(actual).hasValue(expectedDocument);
}
```

Test result output:
 
<img width="1317" alt="Screenshot 2022-05-19 at 17 44 31" src="https://user-images.githubusercontent.com/15107611/169324241-7a2ed65d-e646-4d1c-948d-09d64a74936b.png">

**Comment by roimenashe** [comment]
@Kadavar32 
The NullPointerException is happening because there are no bins when reaching `AerospikeTemplate.putAndGetHeader()` which causes `newRecord` variable in `AerospikeTemplate.doPersistWithVersionAndHandleCasError()` to be null.

Btw starting at version 3.1.0 we still store the user key, the change was that we used to store the key twice - under `@user_key` bin and under `PK`, we no longer store the key under `@user_key`. check out version 3.1.0 release notes: https://github.com/aerospike-community/spring-data-aerospike/releases/tag/3.1.0

Once I changed your object model and your test to include at least 1 bin (in this case `desc`):

Object Model:
```java
@Document
@EqualsAndHashCode
@ToString
@Builder(toBuilder = true)
@RequiredArgsConstructor(onConstructor = @__({@PersistenceConstructor}))
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Data
public class NoWritableFieldsDocument {

    @Id
    public final String id;

    @Version
    public long version;

    @Field("desc")
    public String description;
}
```

Test:
```java
@Test
public void shouldSaveDocumentWithNoWritableFields() {
    NoWritableFieldsDocument expectedDocument = NoWritableFieldsDocument.builder()
            .id("document-key-1")
            .description("description1")
            .build();

    noWritableFieldsDocumentRepository.save(expectedDocument);

    Optional<NoWritableFieldsDocument> actual = noWritableFieldsDocumentRepository.findById(expectedDocument.getId());

    assertThat(actual).hasValue(expectedDocument);
}
```

Test passes and no NPE is raised.
![image](https://user-images.githubusercontent.com/33356310/169689368-f5b3aabd-8da7-488b-bec0-322342ff9135.png)

**Comment by Kadavar32** [comment]
@roimenashe 
Thanks for your reply!.
I know that adding any writable field (like description in your example) will help. Also sending @class bin will help as well (that was I used in my project).

I saw your issue "sending user key twice": PR https://github.com/aerospike-community/spring-data-aerospike/pull/275 and it helped me to debug the issue and understand exact conditions to reproduce the error.
However I understand that the case with document without writable fields is rare enough, but the issue with NPE can be confusing, so maybe it is possible to improve this? As an option custom exception instead NPE if there are no bins when reaching `AerospikeTemplate.putAndGetHeader()` ?
